message_processor.go

```diff
@@ -50,36 +83,48 @@ func consumeLog(lf *logPair) {
 			logrus.WithField("id", lf.info.ContainerID).WithError(err).Debug("Ignoring error")
 			dec = protoio.NewUint32DelimitedReader(lf.stream, binary.BigEndian, 1e6)
 		}
-		if sendMessage(lf.splunkl, &buf, lf.info.ContainerID) == false {
-			continue
-		}
-		if sendMessage(lf.jsonl, &buf, lf.info.ContainerID) == false {
-			continue
-		}
 
+		if shouldSendMessage(buf.Line) {
+			// Append to temp buffer
+			appendToTempBuffer(tmpBuf, &buf)
+			// Check for temp buffer timer expiration
+			diff := time.Now().Sub(tmpBuf.bufferTimer)
```

Comment:
refactor out this as a member function of `tmpBuffer`

```
func (b *tmpBuffer) TimedOut() bool {
   diff := time.Now().Sub(b.bufferTimer)
   if differ > tempMsgBufferHoldDuration {
       return true
   }
    return false
}

```

Reply:
Done